simple_history/tests/tests/test_models.py

```diff
@@ -370,6 +370,29 @@ def test_specify_history_user(self):
             [d.history_user for d in document.history.all()], [None, user2, user1]
         )
 
+    def test_specify_history_user_self_reference_delete(self):
+        user1 = User.objects.create_user("user1", "1@example.com")
+        user2 = User.objects.create_user("user2", "1@example.com")
+        document = Document.objects.create(changed_by=user1)
+        document.changed_by = user2
+        document.save()
+        document.changed_by = None
+        document.save()
+        self.assertEqual(
+            [d.history_user for d in document.history.all()], [None, user2, user1]
+        )
+
+        # Change back to user1
+        document.changed_by = user1
+        document.save()
+
+        # Deleting user1 will cascade delete the document,
+        # but fails when it tries to make the historical
+        # record for the deleted user1.
+        # This test performs differently on Postgres vs. SQLite
+        # because of how the two database handle database constraints
+        user1.delete()
```

Comment:
What's the particular exception you're testing won't be thrown here? When testing that an exception won't be thrown, I generally like to follow this pattern:

```
try:
    the_call_that_will_not_raise_exception()
except ErrorThatWillNotBeRaised as e:
    self.fail(e)
```
```